optimum/neuron/modeling_decoder.py

```diff
@@ -0,0 +1,208 @@
+
+    Common attributes:
+        - model (`torch.nn.Module`) -- The decoder model with a graph optimized for neuron devices.
+        - config ([`~transformers.PretrainedConfig`]) -- The configuration of the original model.
+        - generation_config ([`~transformers.GenerationConfig`]) -- The generation configuration used by default when calling `generate()`.
+    """
+
+    CHECKPOINT_DIR = "checkpoint"
+    COMPILED_DIR = "compiled"
+
+    def __init__(
+        self,
+        model: torch.nn.Module,
+        config: "PretrainedConfig",
+        model_path: Union[str, Path, TemporaryDirectory],
+        generation_config: Optional[GenerationConfig] = None,
+    ):
+        if not is_transformers_neuronx_available() or not isinstance(model, NeuronxPretrainedModel):
+            raise ValueError("The source model must be a transformers_neuronx.PreTrainedModel.")
+
+        super().__init__(model, config)
+        self.model_path = model_path
+        if generation_config is None:
+            generation_config = GenerationConfig.from_model_config(config)
+        self.generation_config = generation_config
+        self.device = torch.device("cpu")
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        config: "PretrainedConfig",
+        use_auth_token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: Optional[str] = None,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        trust_remote_code: bool = False,
+        task: Optional[str] = None,
+        **kwargs,
+    ) -> "NeuronDecoderModel":
+        if not is_transformers_neuronx_available():
+            raise ModuleNotFoundError("The transformers_neuronx package is required to export the model.")
+
+        if task is None:
+            task = TasksManager.infer_task_from_model(cls.auto_model_class)
+
+        # Instantiate the exporter for the specified configuration and task
+        exporter = get_exporter(config, task)
+
+        # Split kwargs between model and neuron args
+        model_kwargs, neuron_kwargs = exporter.split_kwargs(**kwargs)
+
+        # Instantiate the transformers model checkpoint
+        model = TasksManager.get_model_from_task(
+            task=task,
+            model_name_or_path=model_id,
+            subfolder=subfolder,
+            revision=revision,
+            framework="pt",
+            cache_dir=cache_dir,
+            use_auth_token=use_auth_token,
+            local_files_only=local_files_only,
+            force_download=force_download,
+            trust_remote_code=trust_remote_code,
+            **model_kwargs,
+        )
+
+        # Save the model checkpoint in a temporary directory
+        checkpoint_dir = TemporaryDirectory()
+        save_pretrained_split(model, checkpoint_dir.name)
+
+        # Update the config
+        config.neuron = {"task": task, "neuron_kwargs": neuron_kwargs}
```

Comment:
Not in favor of that because we want to keep the model config similar to what is used in Transformers.

Reply:
I need to keep track of the task (to get the correct exporter), then I also need the named parameters that were used to configure the export.

I saw in NeuronBaseModel that the config is already extended with a lot of items, all named `neuron_XX`.
Do you mean you don't want to store anything neuron-specific in the config ? i certainly could serialize the neuron_config in its own file, like the generation_config, but since @JingyaHuang already had put a lot of stuff there I thought it was ok.

Reply:
I did not know. I think it's better to have everything in another file yes.
Do not want to extend the work on this but yes, in general we try not to change the original config.

Reply:
I previously wanted to have a separate config for neuron-specific things, but to move fast I put them under pretrained config. @philschmid 

I now actually think it's not bad to have them in one place.

Reply:
We need to decide for one solution or the other. I don't have any strong opinion myself, but if I put the neuron config in a separate file it will be just for these models for now.

Reply:
I'll stick with adding a neuron config inside the main config (@JingyaHuang did the same for othe rmodels).